services/kuber/server/nodes/nodes.go

```diff
@@ -158,27 +167,79 @@ func (d *Deleter) updateClusterData() {
 
 // deleteFromLonghorn will delete node from nodes.longhorn.io
 // return nil if successful, error otherwise
-func (d *Deleter) deleteFromLonghorn(kc kubectl.Kubectl) error {
-	for _, nodeName := range d.workerNodes {
-		log.Info().Msgf("Deleting node %s from nodes.longhorn.io", nodeName)
-		if err := kc.KubectlDeleteResource("nodes.longhorn.io", nodeName, "longhorn-system"); err != nil {
-			return fmt.Errorf("error while deleting node %s from nodes.longhorn.io : %w", nodeName, err)
+func deleteFromLonghorn(kc kubectl.Kubectl, worker string) error {
+	log.Info().Msgf("Deleting node %s from nodes.longhorn.io", worker)
+	if err := kc.KubectlDeleteResource("nodes.longhorn.io", worker, "-n", longhornNamespace); err != nil {
+		return fmt.Errorf("error while deleting node %s from nodes.longhorn.io : %w", worker, err)
+	}
+	return nil
+}
+
+// assureReplication tries to assure, that replicas for each longhorn volume are migrated to nodes, which will remain in the
```

Comment:
I cannot figure out the ending of this sentence..
regardless, it should probably be added.